integration/tests/karpenter/karpenter_test.go

```diff
@@ -0,0 +1,83 @@
+package karpenter
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/weaveworks/eksctl/integration/runner"
+	"github.com/weaveworks/eksctl/integration/tests"
+	clusterutils "github.com/weaveworks/eksctl/integration/utilities/cluster"
+	"github.com/weaveworks/eksctl/integration/utilities/kube"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/karpenter"
+	"github.com/weaveworks/eksctl/pkg/testutils"
+)
+
+var params *tests.Params
+
+func init() {
+	testing.Init()
+	if err := api.Register(); err != nil {
+		panic(fmt.Errorf("unexpected error registering API scheme: %w", err))
+	}
+	params = tests.NewParams("")
+}
+
+func TestKarpenter(t *testing.T) {
+	testutils.RegisterAndRun(t)
+}
+
+var _ = Describe("(Integration) Karpenter", func() {
+
+	var (
+		clusterName string
+	)
+
+	BeforeEach(func() {
+		clusterName = fmt.Sprintf("it-karpenter-%d", time.Now().Unix())
```

Comment:
Any reason why you are not using the generated `params.ClusterName`? Are you planning to create multiple clusters in this test suite?

Reply:
Too long. :/ Errors out on the Role name creation scheme.

Reply:
It might be worth adding a comment here because most of the other tests rely on the cluster name generated by `tests.NewParams` :slightly_smiling_face: